### PR TITLE
v3

### PR DESCRIPTION
**Pull request by ArnoldZokas**

### CODE REVIEW

File: src/plugin.js [hunk 1]
```diff
@@ -1,141 +1,141 @@
...
 
-            if(whitelist.indexOf(header) > -1) {
-                result.push(header + '=' + requestHeaders[header]);
+            const cacheKey = cacheKeyGenerator.generateCacheKey(req, options);
+
+            if(client.connected ) {
+                try {
+                    client.get(cacheKey, (err, data) => {
+                        if(err) {
+                            return reply.continue();
```
Comment (jwhayes) [inline src/plugin.js]: Do we want to log here?

Comment (ArnoldZokas) [inline src/plugin.js]: good point

**Comment by jwhayes** [comment]
LGTM. As discussed, some ideas for manual tests around reconnection scenarios / general resilience during Redis outages:

1. (Happy path) Start Hapi app with Redis already running. Verify responses are served from cache. 

2a. Start Hapi app without Redis running. Verify responses are served, bypassing cache. 
2b. Start Redis. Verify responses are served from cache. 

3a. Start Hapi app with Redis already running. 
3b. Stop Redis. Verify responses are served, bypassing cache. 
3c. Start Redis. Verify responses are served from cache.

**Comment by ArnoldZokas** [comment]
@jwhayes test complete. No issues surfaced.

**Comment by matteofigus** [comment]
LGTM

**Pull request merged**